Add ri

RJMW opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team](https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team) using a special command in
a comment on the PR to get the attention of the `staged-recipes` team. You can
also consider asking on our [Gitter channel](https://gitter.im/conda-forge/conda-forge.github.io)
or on our [Keybase chat](https://keybase.io/team/condaforge.chat)
if your recipe isn't reviewed promptly.
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/ri```) and found it was in an excellent condition.

RJMW commented:
@conda-forge/core - Please, any suggestions on how to fix the Windows build.

scopatz commented on recipes/ri/bld.bat:
```diff
@@ -0,0 +1,8 @@
+mv "%RECIPE_DIR%\sys" "%SRC_DIR%\sys" 
+sed -i '66s/.*/#include \"sys\/time.h\"/' include\timer.h
```
you probably don't have sed on windows

RJMW replied:
@scopatz - I have replaced the sed command in the .bat file with a patch (e617868).

scopatz commented:
You are getting an error in compilation: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=192766&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=26d9c833-f81c-502c-87bc-0cb0be44f838&l=792

RJMW commented:
> You are getting an error in compilation: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=192766&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=26d9c833-f81c-502c-87bc-0cb0be44f838&l=792

@scopatz - I have fixed this. The current error is identical to the error we got before I switched from sed to patch.

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=192811&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=26d9c833-f81c-502c-87bc-0cb0be44f838&l=804

scopatz commented:
Yeah, I am not sure.  You could skip windows for now and re-enable it in the feedstock later? It seems like this project might not build on windows (or at least not using make?)

RJMW commented:
@scopatz I have simplified the recipe to unix only. Please, can you check if any additional changes are required.

scopatz commented on recipes/ri/meta.yaml:
```diff
@@ -0,0 +1,35 @@
+
+test:
+  commands:
+    - which ri36
```
We usually use `test -f ${PREFIX}/bin/ri36` here

RJMW replied:
Done

scopatz commented:
@conda-forge/staged-recipes - is anyone else seeing the following on Mac?

```
+ setup_conda_rc ./ ./recipes ./.ci_support/osx64.yaml
Configuring conda.
Traceback (most recent call last):
  File "/usr/local/miniconda/bin/setup_conda_rc", line 7, in <module>
    from conda_forge_ci_setup.build_utils import setup_conda_rc
ModuleNotFoundError: No module named 'conda_forge_ci_setup'
```

Also @RJMW - try merging from master

RJMW commented:
@conda-forge-admin, please rerender

RJMW commented:
@scopatz is this ready to merge?

scopatz commented:
Thanks!
